sdk/core/core/test/main.c

```diff
@@ -330,6 +330,12 @@ int main()
     az_json_token token;
     TEST_ASSERT(az_json_parser_parse_token(&state, &token) == AZ_ERROR_BUFFER_OVERFLOW);
   }
+  {
+    az_json_parser state = { 0 };
+    TEST_EXPECT_SUCCESS(az_json_parser_init(&state, AZ_SPAN_FROM_STR("10000000000000000000000e17")));
```

Comment:
Please beef up these tests before checking in.  pos/neg exp, boundary conditions, exp == 0, 1, -1, ....

Reply:
we already had test for that.
But, I am taking the chance to refactor tests files for easier understanding